src/controllers/user.js

```diff
@@ -87,8 +87,20 @@ const getSingleUser = async (req, res) => {
   }
 };
 
+const deleteProfile = async (req, res) => {
+  try {
+    await UserModel.findByIdAndDelete(req.user.userId);
+    await ProductModel.deleteMany({ publisher: req.user.userId });
```

Comment:
can you also add "isTransactionCompleted: false" after the publisher part? i think we should not delete the products that are already given or taken.